src/Sarif.Viewer.VisualStudio/CodeAnalysisResultManager.cs

```diff
@@ -189,77 +201,59 @@ public bool TryResolveFilePath(int resultId, int runIndex, string uriBaseId, str
-                            {
-                                this.AddAllowedDownloadHost(uri.Host);
-                            }
-                        }
-                    }
+            if (string.IsNullOrEmpty(resolvedPath))
+            {
+                // resolve path, existing file in local disk
+                resolvedPath = this.GetRebaselinedFileName(uriBaseId, relativePath, dataCache, solutionPath);
+            }
 
-                    if (allow)
-                    {
-                        try
-                        {
-                            resolvedPath = this.DownloadFile(sarifErrorListItem, uri.ToString());
-                        }
-                        catch (WebException wex)
-                        {
-                            VsShellUtilities.ShowMessageBox(ServiceProvider.GlobalProvider,
-                                       Resources.DownloadFail_DialogMessage + Environment.NewLine + wex.Message,
-                                       null, // title
-                                       OLEMSGICON.OLEMSGICON_CRITICAL,
-                                       OLEMSGBUTTON.OLEMSGBUTTON_OK,
-                                       OLEMSGDEFBUTTON.OLEMSGDEFBUTTON_FIRST);
-                            return false;
-                        }
-                    }
-                }
-                else
+            // verify resolved file with artifact's Hash
+            if (hasHash)
+            {
+                string currentResolvedPath = resolvedPath;
+                if (!this.VerifyFileWithArtifactHash(sarifErrorListItem, relativePath, dataCache, currentResolvedPath, embeddedTempFilePath, out resolvedPath))
                 {
-                    string solutionPath = null;
-                    DTE2 dte = (DTE2)Package.GetGlobalService(typeof(DTE));
-                    if (dte.Solution != null && dte.Solution.IsOpen)
-                    {
-                        solutionPath = dte.Solution.FullName;
-                    }
-
-                    // User needs to locate file.
-                    resolvedPath = this.GetRebaselinedFileName(sarifErrorListItem, uriBaseId, relativePath, dataCache, solutionPath);
+                    return false;
                 }
+            }
 
-                if (string.IsNullOrEmpty(resolvedPath) || relativePath.Equals(resolvedPath, StringComparison.OrdinalIgnoreCase))
+            if (string.IsNullOrEmpty(resolvedPath))
+            {
+                // User needs to locate file.
+                resolvedPath = this._promptForResolvedPathDelegate(sarifErrorListItem, relativePath);
+            }
+
+            if (!string.IsNullOrEmpty(resolvedPath))
+            {
+                // save resolved path to mapping
+                if (!this.SaveResolvedPathToUriBaseMapping(uriBaseId, relativePath, relativePath, resolvedPath, dataCache))
                 {
-                    return false;
+                    resolvedPath = relativePath;
```

Comment:
I think we just need to return false?

Reply:
theoretically yes. this is result of refactoring a big function into couple small functions.
original function returned resolvedPath, and used below if condition to check its value. So here I kept the same style, let each small function set resolvedPath and use below if condition to check and return.

---
In reply to: [608331090](https://github.com/microsoft/sarif-visualstudio-extension/pull/384#discussion_r608331090) [](ancestors = 608331090)